### PR TITLE
Fix failed test/buildbot/builders/unified_cmakeex.py test (issue #235).

### PR DESCRIPTION
**Pull request by vvereschaka**
The changes for UnifiedTreeBuilder.getCmakeExBuildFactory factory and its test:

* updated the step formatting with 'hint' argument (avoid usage of substitutions).
* allowed only 'str' or None for the 'hint' argument.
* updated 'test/buildbot/builders/unified_cmakeex.py' test for the factory changes accordingly.

See #235 for more details.

### CODE REVIEW

**Review by asb — APPROVED** [review]
LGTM in that the test now passes, and to someone not deeply familiar with the code nothing stands out as questionable. Especially given the status quo is that the test doesn't work at all I think this LGTM is probably sufficient, but if you have doubts about and your changes best to wait for additional review.

**Review by gkistanova — APPROVED** [review]
LGTM.

**Pull request merged**
